<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="JReceveur" />

<meta name="date" content="2017-06-11" />

<title>Creating a phylogeny from the 1000 Genome Database</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlightingOnLoad(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Benbow Lab</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Phylogeny.html">Phylogeny from VCF file</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Research Focuses
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Aquatics.html">Aquatic Ecosystems</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Creating a phylogeny from the 1000 Genome Database</h1>
<h4 class="author"><em>JReceveur</em></h4>
<h4 class="date"><em>June 11, 2017</em></h4>

</div>


<div id="creating-a-dendrogram-from-a-downloaded-vcf-file" class="section level2">
<h2>Creating a Dendrogram from a downloaded VCF file</h2>
<p>This document provides an example of creating a dendrogram from a VCF file. The VCF file can be altered using VCFtools to look at a specific area of the genome. The lines in red are the code you should enter, while the lines starting with ## are what you would expect to see for the output</p>
<p>This document was created using Rmarkdown <a href="http://rmarkdown.rstudio.com" class="uri">http://rmarkdown.rstudio.com</a>.</p>
<p>```</p>
</div>
<div id="installing-packages" class="section level1">
<h1>Installing Packages</h1>
<pre class="r"><code>#source(&quot;https://bioconductor.org/biocLite.R&quot;)
#biocLite(&quot;SNPRelate&quot;)</code></pre>
</div>
<div id="loading-packages" class="section level1">
<h1>Loading Packages</h1>
<p>The packages will have to be loaded each time you open R or Rstudio.</p>
<pre class="r"><code>library(gdsfmt)
library(SNPRelate)</code></pre>
<pre><code>## SNPRelate -- supported by Streaming SIMD Extensions 2 (SSE2)</code></pre>
<pre class="r"><code>library(ggplot2)</code></pre>
</div>
<div id="loading-vcf-file" class="section level1">
<h1>Loading VCF file</h1>
<p>There are two ways that the VCF file can be loaded into Rstudio. The first is directly specifying the file path and name, the second is using the file.choose option which will open a browser so you can search for the file. The second optio that is currently commented out with # (putting a # before any code means it will be treated as text, to run it, just remove the #).</p>
<pre class="r"><code>vcf.fn &lt;- &quot;C:\\Users\\Joe Receveur\\Documents\\Virtual Box\\Benin\\10kbExampleData.vcf&quot;
#vcf.fn&lt;- file.choose()</code></pre>
</div>
<div id="parsing" class="section level1">
<h1>Parsing</h1>
<p>The next command will turn the VCF file into a less data intensive form (GDS) for easier computing. If you have loaded the entire genome, expect this command to take an hour or more.</p>
<pre class="r"><code>snpgdsVCF2GDS(vcf.fn,&quot;data.gds&quot;,method =&quot;biallelic.only&quot;)</code></pre>
<pre><code>## VCF Format ==&gt; SNP GDS Format
## Method: exacting biallelic SNPs
## Number of samples: 16
## Parsing &quot;C:\Users\Joe Receveur\Documents\Virtual Box\Benin\10kbExampleData.vcf&quot; ...
##  import 2148 variants.
## + genotype   { Bit2 16x2148, 8.4K } *
## Optimize the access efficiency ...
## Clean up the fragments of GDS file:
##     open the file &#39;data.gds&#39; (25.5K)
##     # of fragments: 39
##     save to &#39;data.gds.tmp&#39;
##     rename &#39;data.gds.tmp&#39; (25.3K, reduced: 228B)
##     # of fragments: 20</code></pre>
</div>
<div id="formatting-for-dissimilarity-matrix" class="section level1">
<h1>Formatting for dissimilarity matrix</h1>
<p>These commands prepare the data so it is formatted correctly to create a dissimilarity matrix.</p>
<pre class="r"><code>genofile&lt;-snpgdsOpen(&quot;data.gds&quot;)
set.seed(100)
ibs.hc&lt;-snpgdsHCluster(snpgdsIBS(genofile,num.thread=2, autosome.only=FALSE))</code></pre>
<pre><code>## Identity-By-State (IBS) analysis on genotypes:
## Excluding 46 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
## Working space: 16 samples, 2,102 SNPs
##     using 2 (CPU) cores
## IBS:    the sum of all selected genotypes (0,1,2) = 48494
## Fri Nov 10 15:30:03 2017    (internal increment: 65536)
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed in 0s
## Fri Nov 10 15:30:03 2017    Done.</code></pre>
</div>
<div id="turn-the-clustering-into-a-tree-file-and-plotting-tree" class="section level1">
<h1>Turn the clustering into a tree file and plotting tree</h1>
<p>This step takes the clustering results from before and turns the numerical values into a dendrogram</p>
<pre class="r"><code>rv &lt;- snpgdsCutTree(ibs.hc)</code></pre>
<pre><code>## Determine groups by permutation (Z threshold: 15, outlier threshold: 5):
## Create 1 groups.</code></pre>
<pre class="r"><code>plot(rv$dendrogram,main=&quot;Within 10 kb of ACE-1&quot;)</code></pre>
<p><img src="Phylogeny_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
</div>
<div id="dissimilarity-matrix" class="section level1">
<h1>Dissimilarity matrix</h1>
<p>This command creates a dissimilarity matrix between all the samples. If you are looking at the X chromosome, make sure the autosome.only= code is changed to autosome.only=False.</p>
<pre class="r"><code>dissMatrix  =  snpgdsIBS(genofile , sample.id=NULL, autosome.only=TRUE,remove.monosnp=TRUE,  maf=NaN, missing.rate=NaN, num.thread=2, verbose=TRUE)</code></pre>
<pre><code>## Identity-By-State (IBS) analysis on genotypes:
## Excluding 0 SNP on non-autosomes
## Excluding 46 SNPs (monomorphic: TRUE, MAF: NaN, missing rate: NaN)
## Working space: 16 samples, 2,102 SNPs
##     using 2 (CPU) cores
## IBS:    the sum of all selected genotypes (0,1,2) = 48494
## Fri Nov 10 15:30:03 2017    (internal increment: 65536)
## 
[..................................................]  0%, ETC: ---    
[==================================================] 100%, completed in 0s
## Fri Nov 10 15:30:03 2017    Done.</code></pre>
</div>
<div id="clustering-analysis" class="section level1">
<h1>Clustering Analysis</h1>
<p>This step performs a clustering analysis similar to above but with a different equation.The next line creates a tree file based on dissimilarity rather than relatedness.</p>
<pre class="r"><code>snpHCluster =  snpgdsHCluster(dissMatrix, sample.id=NULL, need.mat=TRUE, hang=0.01)
cutTree = snpgdsCutTree(snpHCluster, z.threshold=15, outlier.n=5, n.perm = 5000, samp.group=NULL, 
                        col.outlier=&quot;red&quot;, col.list=NULL, pch.outlier=4, pch.list=NULL,label.H=FALSE, label.Z=TRUE, 
                        verbose=TRUE)</code></pre>
<pre><code>## Determine groups by permutation (Z threshold: 15, outlier threshold: 5):
## Create 1 groups.</code></pre>
<pre class="r"><code>cutTree</code></pre>
<pre><code>## $sample.id
##  [1] &quot;AA0035-C&quot; &quot;AA0036-C&quot; &quot;AA0037-C&quot; &quot;AA0038-C&quot; &quot;AB0286-C&quot; &quot;AB0290-C&quot;
##  [7] &quot;AB0292-C&quot; &quot;AX0001-C&quot; &quot;AX0002-C&quot; &quot;AY0001-C&quot; &quot;AY0002-C&quot; &quot;BY0001-C&quot;
## [13] &quot;BY0002-C&quot; &quot;BZ0001-C&quot; &quot;BZ0003-C&quot; &quot;BZ0004-C&quot;
## 
## $z.threshold
## [1] 15
## 
## $outlier.n
## [1] 5
## 
## $samp.order
##  [1] 14  1  3  2  4 16 15  7 11 10  5  6  9 12  8 13
## 
## $samp.group
##  [1] G001 G001 G001 G001 G001 G001 G001 G001 G001 G001 G001 G001 G001 G001
## [15] G001 G001
## Levels: G001
## 
## $dmat
##           G001
## G001 0.2068935
## 
## $dendrogram
## &#39;dendrogram&#39; with 2 branches and 16 members total, at height 0.2700422 
## 
## $merge
##            z n1 n2
## 1  0.0000000  1  1
## 2  1.3070571  1  2
## 3  1.7842348  1  3
## 4  0.0000000  1  1
## 5  0.0000000  1  1
## 6  0.7506769  1  2
## 7  1.5654121  2  3
## 8  1.6302752  1  5
## 9  2.0057114  1  4
## 10 0.0000000  1  1
## 11 1.4528401  1  2
## 12 2.4221257  1  6
## 13 6.1503264  5  7
## 14 1.6738589  1  3
## 15 5.5572832 12  4
## 
## $clust.count
## G001 
##   16</code></pre>
<pre class="r"><code>snpgdsClose(genofile)</code></pre>
</div>
<div id="displaying-a-tree-based-on-dissimilarity" class="section level1">
<h1>Displaying a tree based on dissimilarity</h1>
<p>Even though the tree is based on dissimilarity, the closest samples are still the most similar or to be more accurate, least dissimilar.</p>
<pre class="r"><code>snpgdsDrawTree(cutTree, main = &quot;Phylogenetic Tree&quot;,edgePar=list(col=rgb(0.5,0.5,0.5,0.75),t.col=&quot;black&quot;),
               y.label.kinship=T,leaflab=&quot;perpendicular&quot;)</code></pre>
<p><img src="Phylogeny_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.4.2 (2017-09-28)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 14393)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_United States.1252 
## [2] LC_CTYPE=English_United States.1252   
## [3] LC_MONETARY=English_United States.1252
## [4] LC_NUMERIC=C                          
## [5] LC_TIME=English_United States.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggplot2_2.2.1    SNPRelate_1.12.0 gdsfmt_1.14.0   
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_0.12.13     digest_0.6.12    rprojroot_1.2    plyr_1.8.4      
##  [5] grid_3.4.2       gtable_0.2.0     backports_1.1.1  magrittr_1.5    
##  [9] scales_0.5.0     evaluate_0.10.1  rlang_0.1.4      stringi_1.1.5   
## [13] lazyeval_0.2.1   rmarkdown_1.7    tools_3.4.2      stringr_1.2.0   
## [17] munsell_0.4.3    yaml_2.1.14      compiler_3.4.2   colorspace_1.3-2
## [21] htmltools_0.3.6  knitr_1.17       tibble_1.3.4</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
